File: src/test/java/org/micoli/php/builders/ShellBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.Shell;

public class ShellBuilder {
    private final Shell shell;

    private ShellBuilder() {
        this.shell = new Shell();
    }

    public static ShellBuilder create() {
        return new ShellBuilder();
    }

    public ShellBuilder withId(String id) {
        shell.id = id;
        return this;
    }

    public ShellBuilder withLabel(String label) {
        shell.label = label;
        return this;
    }

    public ShellBuilder withCommand(String command) {
        shell.command = command;
        return this;
    }

    public ShellBuilder withCwd(String cwd) {
        shell.cwd = cwd;
        return this;
    }

    public ShellBuilder withIcon(String icon) {
        shell.icon = icon;
        return this;
    }

    public Shell build() {
        return shell;
    }
}

File: src/test/kotlin/org/micoli/php/builders/ShellBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.Shell

class ShellBuilder private constructor() {
    private val shell: Shell = Shell()

    fun withId(id: String): ShellBuilder {
        shell.id = id
        return this
    }

    fun withLabel(label: String): ShellBuilder {
        shell.label = label
        return this
    }

    fun withCommand(command: String): ShellBuilder {
        shell.command = command
        return this
    }

    fun withCwd(cwd: String): ShellBuilder {
        shell.cwd = cwd
        return this
    }

    fun withIcon(icon: String): ShellBuilder {
        shell.icon = icon
        return this
    }

    fun build(): Shell = shell

    companion object {
        @JvmStatic fun create(): ShellBuilder = ShellBuilder()
    }
}
